### ISSUE TITLE
MissingShell triggers DuplicateShellOnSteps

### ISSUE THREAD

**Issue by TWiStErRob** [TWiStErRob/net.twisterrob.ghlint#129]
### Version

0.3.1

### Reproduction steps

1. File `.github/workflows/x.yml` with the following content:
   ```yaml
   on: push
   jobs:
     repro:
       runs-on: ubuntu-latest
       steps:
         - run: echo "Problem!"
         - run: echo "Problem!"
   ```

2. Run `ghlint` with the following arguments:
   ```shell
   ghlint --console .github/workflows/x.yml
   ```

### Expected behavior

Helps to get to an end state.

### Actual behavior

MissingShell should recommend default if there's multiple shells.
